PD-374: Add new options for Migration spec (#879)
portworx/pxdocs: content/portworx-install-with-kubernetes/migration/kubemotion/_index.md

Alternatively, you can use your own cloud (S3, Azure, or Google) instead of ObjectStore on the destination cluster. Note that the credentials must be named `clusterPair_<clusterUUID_of_destination>` and you are required to create them on both the source and the destination cluster.
{{</info>}}

## Start a migration

1. Create a file called `migration.yaml` file, specifying the following fields and values:

  * **apiVersion:** as `stork.libopenstorage.org/v1alpha1`
  * **kind:** as `Migration`
  * **metadata.name:** with the name of the object that performs the migration
  * **metadata.namespace:** with the name of the namespace in which you want to create the object
  * **spec.clusterPair:** with the name of the `ClusterPair` object created in the [Create a ClusterPair](#create-a-clusterpair) section

>   * **spec.includeResources:** with a boolean value specifying if the migration should include PVCs and applications <!-- Are there any other resources we should mention?-->. If you set this field to `false`, Portworx will only migrate your volumes

Author: @disrani-px

>   * **spec.includeResources:** with a boolean value specifying if the migration should include PVCs and applications <!-- Are there any other resources we should mention?-->. If you set this field to `false`, Portworx will only migrate your volumes

Reviewer: ```suggestion
  * **spec.includeResources:** with a boolean value specifying if the migration should include PVCs and other applications specs. If you set this field to `false`, Portworx will only migrate your volumes
```

  * **spec.startApplications:** with a boolean value specifying if Portworx should automatically start applications on the destination cluster. If you set this field to `false`, then the `Deployment` and `StatefulSet` objects on the destination cluster will be set to zero. Note that, on the destination cluster, Portworx uses the `stork.openstorage.org/migrationReplicas` annotation to store the number of replicas from the source cluster

>   * **spec.namespaces**: with the list of namespaces Portworx should migrate

Reviewer: ```suggestion
  * **spec.namespaces**: with the list of namespaces to be migrated
```

>   * **spec.adminClusterPair:** with the name of the `ClusterPair` object created in the admin namespace that Portworx should use to migrate your cluster-scoped resources. Use this if your regular users lack permission to create or delete cluster-scoped resources on the destination cluster. If you don't specify this field, then Portworx will use the object specified in the `spec.clusterPair` field. This feature was introduced in STORK v2.3.0.

Reviewer: ```suggestion
  * **spec.adminClusterPair:** with the name of the `ClusterPair` object created in the admin namespace that Portworx should use to migrate your cluster-scoped resources. Use this if your regular users lack permission to create or delete cluster-scoped resources on the destination cluster. If you don't specify this field, then Stork will use the object specified in the `spec.clusterPair` field. This feature was introduced in STORK v2.3.0.
```

>   * **spec.purgeDeletedResources:** with a boolean value specifying if Portworx should automatically purge a resource from the destination cluster when you delete it from the source cluster. The default value is `false`. This feature was introduced in STORK v2.3.2.

Reviewer: ```suggestion
  * **spec.purgeDeletedResources:** with a boolean value specifying if Stork should automatically purge a resource from the destination cluster when you delete it from the source cluster. The default value is `false`. This feature was introduced in STORK v2.3.2.
```
